FreeRTOS/FreeRTOS-Kernel
# Adding FreeRTOS SMP support

Description
-----------
FreeRTOS SMP enable more cores to run on instance of FreeRTOS at the same time. It is developed in the SMP branch. This PR merge the SMP branch to main branch with some modification so that single core FreeRTOS-kernel is not effected.

> Note the unit test in FreeRTOS/FreeRTOS is updated in another PR. This PR should be merged after unit test PR is merged.

Test Steps
-----------
Use RP2040 port with configNUMBER_OF_CORES > 1 to enable FreeRTOS SMP scheduler.

Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
